### PR TITLE
fix: upgrade deprecated github actions

### LINKED ISSUE #14321: [CAL-3356] Upgrade deprecated Github Actions
We starting to get these warnings

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, dorny/paths-filter@v2, buildjet/setup-node@v3, buildjet/cache@v3, ataylorme/eslint-annotate-action@v2. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3356](https://linear.app/calcom/issue/CAL-3356/upgrade-deprecated-github-actions)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Upgrade deprecated github actions 

- Fixes #14321 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

**Comment by keithwillcode** [comment]
@anikdhabal thank you for the PR. 

please update the description so it’s not incomplete

File: .github/workflows/nextjs-bundle-analysis.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@ jobs:
           body="${body//'%'/'%25'}"
           body="${body//$'\n'/'%0A'}"
           body="${body//$'\r'/'%0D'}"
-          echo ::set-output name=body::$body
+          echo "body=$body" >> "$GITHUB_OUTPUT"
```
Comment (keithwillcode) [inline .github/workflows/nextjs-bundle-analysis.yml:109]: I had tried this in the past and it caused major output issues on the PR

Comment (keithwillcode) [reply .github/workflows/nextjs-bundle-analysis.yml:109]: Let’s not include on this PR and instead make a separate PR that targets only this change

**Review by keithwillcode — CHANGES_REQUESTED** [review]

File: .github/workflows/nextjs-bundle-analysis.yml (hunk 1 above)
Comment (anikdhabal) [reply .github/workflows/nextjs-bundle-analysis.yml:109]: sure keith

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**